shared/common-adapters/use-popup.tsx

```diff
@@ -0,0 +1,18 @@
+import * as React from 'react'
+
+export const usePopup = (_ref: React.MutableRefObject<any>, makePopup: () => React.ReactNode) => {
+  const [showingPopup, setShowingPopup] = React.useState(false)
+  const [popup, setPopup] = React.useState<React.ReactNode>(null)
+
+  React.useEffect(() => {
+    if (showingPopup === !popup) {
+      setPopup(showingPopup ? makePopup() : null)
+    }
+  }, [showingPopup, popup, makePopup])
+
+  return {
+    popup,
+    setShowingPopup,
+    showingPopup,
+  }
```

Comment:
Should this returned object be typed?

Reply:
it is typed automatically

Reply:
you rarely ever need to explicitly type a return value

Reply:
even when the function is exported and used elsewhere?

Reply:
yes, just have TS show you the type, it exists